src/qml/pages/node/PeerDetails.qml

```diff
@@ -0,0 +1,111 @@
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import org.bitcoincore.qt 1.0
+import "../../controls"
+import "../../components"
+
+Page {
+    signal backClicked()
+
+    property PeerDetailsModel details
+
+    Connections {
+        target: details
+        function onDisconnected() {
+            root.backClicked()
+        }
+    }
+
+    id: root
+    background: null
+
+    header: NavigationBar2 {
+        leftItem: NavButton {
+            iconSource: "image://images/caret-left"
+            text: qsTr("Back")
+            onClicked: root.backClicked()
+        }
+        centerItem: Header {
+            headerBold: true
+            headerSize: 18
+            header: qsTr("Peer " + details.nodeId)
+        }
+    }
+
+    component PeerKeyValueRow: Row {
+        width: parent.width
+        property string key: ""
+        property string value: ""
+        CoreText {
+            color: Theme.color.neutral9;
+            text: key;
+            width: 125;
+            horizontalAlignment: Qt.AlignLeft;
+        }
+        CoreText {
+            color: Theme.color.neutral9;
+            maximumLineCount: 1;
+            elide: Text.ElideRight;
+            wrapMode: Text.NoWrap;
+            text: value
+            width: parent.width - 125;
+            horizontalAlignment: Qt.AlignLeft;
+        }
+    }
+
+    ScrollView {
+        id: scrollView
+        width: parent.width
+        height: parent.height
+        clip: true
+        contentWidth: width
+
+        Column {
+            width: Math.min(parent.width - 40, 450)
+            anchors.horizontalCenter: parent.horizontalCenter
+            spacing: 20
+
+            CoreText { text: "Information"; bold: true; font.pixelSize: 18; horizontalAlignment: Qt.AlignLeft; }
+            Column {
+                width: parent.width
+                PeerKeyValueRow { key: qsTr("IP"); value: details.address }
+                PeerKeyValueRow { key: qsTr("VIA"); value: details.addressLocal }
+                PeerKeyValueRow { key: qsTr("Type"); value: details.type }
+                PeerKeyValueRow { key: qsTr("Permission"); value: details.permission }
+                PeerKeyValueRow { key: qsTr("Version"); value: details.version }
+                PeerKeyValueRow { key: qsTr("User agent"); value: details.userAgent }
+                PeerKeyValueRow { key: qsTr("Services"); value: details.services }
+                PeerKeyValueRow { key: qsTr("Transport relay"); value: details.transactionRelay }
+                PeerKeyValueRow { key: qsTr("Address relay"); value: details.addressRelay }
+                PeerKeyValueRow { key: qsTr("Mapped AS"); value: details.mappedAS }
+            }
+
+            CoreText { text: "Block data"; bold: true; font.pixelSize: 18; horizontalAlignment: Qt.AlignLeft; }
+            Column {
+                width: parent.width
+                PeerKeyValueRow { key: qsTr("Starting block"); value: details.startingHeight }
```

Comment:
`Starting Block` doesn't seem to work correctly, for me this always shows 0
(testing with a synced signet node, normal bitcoin qt it works fine)

Reply:
I noticed that the Qt gui uses a value from a different struct. I will switch to that and see if that resovles it.

Reply:
yes it works now